core/src/test/scala/unit/kafka/server/MetadataCacheTest.scala

```diff
@@ -128,16 +128,18 @@ class MetadataCacheTest {
         partitionMetadatas.zipWithIndex.foreach { case (partitionMetadata, partitionId) =>
           assertEquals(Errors.NONE, partitionMetadata.error)
           assertEquals(partitionId, partitionMetadata.partition)
-          val leader = partitionMetadata.leader
           val partitionState = topicPartitionStates.find(_.partitionIndex == partitionId).getOrElse(
             Assertions.fail(s"Unable to find partition state for partition $partitionId"))
-          assertEquals(partitionState.leader, leader.id)
+          assertEquals(partitionState.leader, partitionMetadata.leaderId)
           assertEquals(Optional.of(partitionState.leaderEpoch), partitionMetadata.leaderEpoch)
-          assertEquals(partitionState.isr, partitionMetadata.isr.asScala.map(_.id).asJava)
-          assertEquals(partitionState.replicas, partitionMetadata.replicas.asScala.map(_.id).asJava)
-          val endpoint = endpoints(partitionMetadata.leader.id).find(_.listener == listenerName.value).get
-          assertEquals(endpoint.host, leader.host)
-          assertEquals(endpoint.port, leader.port)
+          assertEquals(partitionState.isr, partitionMetadata.inSyncReplicaIds)
+          assertEquals(partitionState.replicas, partitionMetadata.replicaIds)
+
+            // TODO: Figure out what to do with these checks which no longer make sense
+//          val endpoint = endpoints(partitionMetadata.leaderId).find(_.listener == listenerName.value).get
+//          val leader = partitionMetadata.leader
+//          assertEquals(endpoint.host, leader.host)
+//          assertEquals(endpoint.port, leader.port)
```

Comment:
Let's figure it out. :)

Reply:
I will just get rid of these checks. I think they are just there to assert consistency between the Node information stored at the partition level in the old code. The new logic does not give the opportunity for such an inconsistency.